velox/common/memory/SharedArbitrator.cpp

```diff
@@ -281,10 +282,22 @@ uint64_t SharedArbitrator::decrementFreeCapacityLocked(
   return allocatedBytes;
 }
 
+uint64_t SharedArbitrator::adjustedShrinkCapacity(
+    const MemoryPool& pool,
+    uint64_t targetBytes) const {
+  const auto reservedBytes = pool.reservedBytes();
+  const auto headroomBytes =
+      std::min(static_cast<uint64_t>(reservedBytes / 4), 128ULL << 20);
```

Comment:
Do we want 128ULL << 20 a control parameter? If it is zero, then we can at least revert to the previous behavior?

Reply:
We can have a separate flag to enable/disable the adjustments. Giving special meaning to one of these parameters could be confusing sometimes.